pkg/scheduler/framework/preemption/preemption_test.go

```diff
@@ -355,3 +359,470 @@ func TestSelectCandidate(t *testing.T) {
+	return f
+}
+
+func (f *fakeExtender) Name() string {
+	return "fakeExtender"
+}
+
+func (f *fakeExtender) IsIgnorable() bool {
+	return f.ignorable
+}
+
+func (f *fakeExtender) ProcessPreemption(
+	_ *v1.Pod,
+	victims map[string]*extenderv1.Victims,
+	_ framework.NodeInfoLister,
+) (map[string]*extenderv1.Victims, error) {
+	if f.isSupportsPreemption {
+		if f.errProcessPreemption {
+			return nil, errors.New("extender preempt error")
+		}
+		if f.isNoReturnVictims {
+			return map[string]*extenderv1.Victims{"mock": {}}, nil
+		}
+		return victims, nil
+	}
+	return nil, nil
+}
+
+func (f *fakeExtender) SupportsPreemption() bool {
+	return f.isSupportsPreemption
+}
+
+func (f *fakeExtender) IsInterested(pod *v1.Pod) bool {
+	return pod != nil
+}
+
+func (f *fakeExtender) Filter(_ *v1.Pod, _ []*framework.NodeInfo) ([]*framework.NodeInfo, extenderv1.FailedNodesMap, extenderv1.FailedNodesMap, error) {
+	return nil, nil, nil, nil
+}
+
+func (f *fakeExtender) Prioritize(
+	_ *v1.Pod,
+	_ []*framework.NodeInfo,
+) (hostPriorities *extenderv1.HostPriorityList, weight int64, err error) {
+	return nil, 0, nil
+}
+
+func (f *fakeExtender) Bind(_ *v1.Binding) error {
+	return nil
+}
+
+func (f *fakeExtender) IsBinder() bool {
+	return true
+}
+
+func (f *fakeExtender) IsPrioritizer() bool {
+	return true
+}
+
+func (f *fakeExtender) IsFilter() bool {
+	return true
+}
+
+func TestCallExtenders(t *testing.T) {
+	var (
+		node1Name            = "node1"
+		defaultSchedulerName = "default-scheduler"
+	)
+	var (
+		preemptor = st.MakePod().Name("preemptor").UID("preemptor").
+				SchedulerName(defaultSchedulerName).Priority(highPriority).
+				Containers([]v1.Container{st.MakeContainer().Name("container1").Obj()}).
+				Obj()
+		victim = st.MakePod().Name("victim").UID("victim").
+			Node(node1Name).SchedulerName(defaultSchedulerName).Priority(midPriority).
+			Containers([]v1.Container{st.MakeContainer().Name("container1").Obj()}).
+			Obj()
+		makeCandidates = func(nodeName string, pods []*v1.Pod) []Candidate {
+			return []Candidate{
+				&fakeCandidate{
+					name: nodeName,
+					victims: &extenderv1.Victims{
+						Pods: pods,
+					},
+				},
+			}
+		}
+	)
+	tests := []struct {
+		name           string
+		extenders      []framework.Extender
+		candidates     []Candidate
+		wantStatus     *framework.Status
+		wantCandidates []Candidate
+	}{
+		{
+			name:           "no extenders",
+			extenders:      []framework.Extender{},
+			candidates:     makeCandidates(node1Name, []*v1.Pod{victim}),
+			wantStatus:     nil,
+			wantCandidates: makeCandidates(node1Name, []*v1.Pod{victim}),
+		},
+		{
+			name: "one extender supports preemption",
+			extenders: []framework.Extender{
+				newFakeExtender().WithSupportsPreemption(true),
+			},
+			candidates:     makeCandidates(node1Name, []*v1.Pod{victim}),
+			wantStatus:     nil,
+			wantCandidates: makeCandidates(node1Name, []*v1.Pod{victim}),
+		},
+		{
+			name: "one extenders",
```

Comment:
This test case looks the same as "one extender supports preemption".

Reply:
done